docs/src/main/paradox/durable-state-store.md

```diff
@@ -41,3 +41,92 @@ don't reuse the same revision numbers that have been deleted.
+
+@extref:[Durable state actors](akka:typed/durable-state/persistence.html) can only be looked up by their entity id.
+Additional indexed data can be stored as a query representation. You can either store the query representation from
+an asynchronous @ref:[Projection](projection.md) or you can store it in the same transaction as the Durable State
+upsert or delete.
+
+Advantages of storing the query representation in the same transaction as the Durable State change:
+
+* exactly-once processing and atomic update with the Durable State change
+* no eventual consistency delay from asynchronous Projection processing
+* no need for Projection offset storage
+
+That said, for write heavy Durable State, a Projection can have the advantage of not impacting write latency of
+the Durable State updates. Note that updating the secondary index also has an impact on the write latency.
+
+### Additional columns
+
+In many cases you just want a secondary index on one or a few fields other than the entity id. For that purpose
+you can configure one or more @apidoc[AdditionalColumn] classes for an entity type. The `AdditionalColumn` will
+extract the field from the Durable State value and define how to bind it to a database column.
+
+The configuration:
+
+@@snip [application.conf](/docs/src/test/scala/docs/home/state/BlogPostTitleColumn.scala) { #additional-column-config }
+
+For each entity type you can define a list of fully qualified class names of `AdditionalColumn` implementations.
+The `AdditionalColumn` implementation may optionally define an ActorSystem constructor parameter.
+
+`AdditionalColumn` for a secondary index on the title of @extref:[blog posts (example in the Akka documentation)](akka:typed/durable-state/persistence.html#changing-behavior):
+
+Scala
+:  @@snip [BlogPostTitleColumn.scala](/docs/src/test/scala/docs/home/state/BlogPostTitleColumn.scala) { #additional-column }
+
+Java
+:  @@snip [BlogPostTitleColumn.java](/docs/src/test/java/jdocs/home/state/BlogPostTitleColumn.java) { #additional-column }
+
+From the `bind` method you can return one of:
+
+* @scala[`AdditionalColumn.BindValue`]@java[`AdditionalColumn.bindValue`] - bind a value such as a `String` or `Long` to the database column
+* @scala[`AdditionalColumn.BindNull`]@java[`AdditionalColumn.bindNull`] - store `null` in the database column
+* @scala[`AdditionalColumn.Skip`]@java[`AdditionalColumn.skip`] - don't update the database column for this change, keep existing value
+
+You would have to add the additional columns to the `durable_state` table definition, and create secondary database index.
+Unless you only have one entity type it's best to define a separate table with the `custom-table` configuration, see
+example above. The custom table should have the same table definition as the default `durable_state` table but with the
+extra columns added.
+
+Note that the full serialized state is stored in the same table as the additional columns. That state can be
```

Comment:
Does this mean the full state is duplicated in both the regular table and the custom-table, or only put in the custom table, maybe clarify?